src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -357,6 +400,47 @@ public struct RequestEventDispatcher
 
     /***************************************************************************
 
+        Convenience wrapper around `nextEvent` for yielding only.
+
+        Params:
+            fiber = fiber to suspend until the an event-loop cycle has occurred
+
+    ***************************************************************************/
+
+    public void yield ( MessageFiber fiber )
+    {
+        auto event = this.nextEvent(fiber, Yield());
+        enforce(event.active == event.active.yielded_resumed,
+            "Unexpected event: waiting only for resumption after yield");
+    }
+
+    /***************************************************************************
+
+        Every `yield_after` calls, suspends `fiber` and resumes it on the next
+        event-loop cycle. Otherwise does nothing.
+
+        Params:
+            fiber = fiber to suspend until the an event-loop cycle has occurred
+            call_count = counter incremented each time this method is called
+            yield_after = the number of calls after which this method, when
+                called again, should yield
+
+    ***************************************************************************/
+
+    public void periodicYield ( MessageFiber fiber, ref uint call_count,
+        Const!(uint) yield_after )
+    {
+        if ( call_count >= yield_after )
```

Comment:
Haha, yes, there's one wasted indeed, but saves you few jumps ;-). Not important, really.